perfkitbenchmarker/linux_benchmarks/cloud_spanner_ycsb_benchmark.py

```diff
@@ -0,0 +1,233 @@
+flags.DEFINE_string('cloud_spanner_ycsb_table',
+                    'usertable',
+                    'The table name used in the YCSB benchmark.')
+flags.DEFINE_integer('cloud_spanner_ycsb_zeropadding',
+                     12,
+                     'The zero padding used in the YCSB benchmark.')
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['ycsb_client_vms'].present:
+    config['vm_groups']['default']['vm_count'] = FLAGS.ycsb_client_vms
+  return config
+
+
+def CheckPrerequisites(benchmark_config):
+  for scope in REQUIRED_SCOPES:
+    if scope not in FLAGS.gcloud_scopes:
+      raise ValueError('Scope {0} required.'.format(scope))
+
+
+def Prepare(benchmark_spec):
+  """Prepare the virtual machines to run cloud spanner benchmarks.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  benchmark_spec.always_call_cleanup = True
+
+  benchmark_spec.spanner_instance = gcp_spanner.GcpSpannerInstance(
+      name=FLAGS.cloud_spanner_ycsb_instance,
+      description=FLAGS.cloud_spanner_ycsb_description,
+      nodes=FLAGS.cloud_spanner_ycsb_nodes,
+      database=FLAGS.cloud_spanner_ycsb_database,
+      ddl=FLAGS.cloud_spanner_ycsb_ddl,
+      project=FLAGS.cloud_spanner_ycsb_project,
+      config=FLAGS.cloud_spanner_ycsb_config,
+      end_point=FLAGS.cloud_spanner_ycsb_end_point)
+  if benchmark_spec.spanner_instance._Exists(instance_only=True):
+    logging.warning('Cloud Spanner instance %s exists, delete it first.' %
+                    FLAGS.cloud_spanner_ycsb_instance)
+    benchmark_spec.spanner_instance.Delete()
+  benchmark_spec.spanner_instance.Create()
+  if not benchmark_spec.spanner_instance._Exists():
+    logging.warning('Failed to create Cloud Spanner instance and database.')
+    benchmark_spec.spanner_instance.Delete()
+
+  default_ycsb_tar_url = ycsb.YCSB_TAR_URL
+
+  # TODO: figure out a less hacky way to override.
+  # Override so that we only need to download the required binding.
+  if FLAGS.cloud_spanner_ycsb_custom_release:
+    ycsb.YCSB_TAR_URL = FLAGS.cloud_spanner_ycsb_custom_release
+  else:
+    ycsb.YCSB_TAR_URL = YCSB_BINDING_TAR_URL
+
+  logging.info('YCSB tar url: ' + ycsb.YCSB_TAR_URL)
+
+  vms = benchmark_spec.vms
+
+  # Install required packages and copy credential files
+  vm_util.RunThreaded(_Install, vms)
+
+  # Restore YCSB_TAR_URL
+  ycsb.YCSB_TAR_URL = default_ycsb_tar_url
+  benchmark_spec.executor = ycsb.YCSBExecutor('cloudspanner')
+
+
+def Run(benchmark_spec):
+  """Spawn YCSB and gather the results.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample instances.
+  """
+  vms = benchmark_spec.vms
+  run_kwargs = {
+      'table': FLAGS.cloud_spanner_ycsb_table,
```

Comment:
as mentioned in previous comment. this flag should belong to ycsb.
Otherwise, are you going to make a datastore_ycsb_table, bigtable_ycsb_table, ...?